File: src/main/java/com/example/demo/controller/OwnerController.java
package com.example.demo.controller;
import org.springframework.web.bind.annotation.RestController;

import com.example.demo.model.Owner;
import com.example.demo.repository.OwnerRepository;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;



@RestController
@RequestMapping("/owner")
public class OwnerController {
    private final OwnerRepository ownerRepository;

    public OwnerController(OwnerRepository ownerRepository){
        this.ownerRepository = ownerRepository;
    }
    @GetMapping("{email}")
    public Owner getOwnerByEmail(@PathVariable String email){
        return ownerRepository.findByEmail(email).orElseThrow(() -> new RuntimeException("Owner not found"));
    }
}

File: src/main/java/com/example/demo/controller/OwnerController.kt
package com.example.demo.controller

import com.example.demo.model.Owner
import com.example.demo.repository.OwnerRepository
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.util.function.Supplier

@RestController
@RequestMapping("/owner")
class OwnerController(private val ownerRepository: OwnerRepository) {
    @GetMapping("{email}")
    fun getOwnerByEmail(@PathVariable email: String?): Owner {
        return ownerRepository.findByEmail(email)
    }
}